src/ast/ast.h

```diff
@@ -28,6 +28,7 @@ class Expression : public Node {
   bool is_literal = false;
   bool is_variable = false;
   bool is_map = false;
+  static std::vector<std::string> resolve;
```

Comment:
This could be a static member of the Cast class, to keep it more localised

Reply:
I like the idea though - I think there's some other code in here we could clean up with more use of static members like this

Reply:
> This could be a static member of the Cast class, to keep it more localised

right, will move it there